tests/test_covariance_clusters.py

```diff
@@ -60,94 +63,87 @@ def get_mock_sacc():
     return s
 
 
-def get_mock_covariance():
+@pytest.fixture
+def mock_covariance(mock_sacc, mock_cosmo):
 
     cc_cov = ClusterCounts(input_yml)
-    cc_cov.load_from_sacc(get_mock_sacc())
-    cc_cov.load_from_cosmology(get_mock_cosmo())
+    cc_cov.load_from_sacc(mock_sacc)
+    cc_cov.load_from_cosmology(mock_cosmo)
     cc_cov.fft_helper = FFTHelper(
-        get_mock_cosmo(), cc_cov.z_lower_limit, cc_cov.z_upper_limit
+        mock_cosmo, cc_cov.z_lower_limit, cc_cov.z_upper_limit
     )
-    cc_cov.mass_func = hmf.MassFuncTinker10(get_mock_cosmo())
+    cc_cov.mass_func = hmf.MassFuncTinker10(mock_cosmo)
     cc_cov.h0 = 0.67
     return cc_cov
 
 
-def test_integral_mass_no_bias():
-    ref1 = 1.463291259900985e-05
-    ref2 = 1.4251538328691035e-05
-    cc_cov = get_mock_covariance()
+@pytest.mark.parametrize(
+    "z, ref_val",
+    [
+        (0.3, 1.463291259900985e-05),
+        (0.35, 1.4251538328691035e-05),
+    ],
+)
+def test_integral_mass_no_bias(mock_covariance: ClusterCounts, z, ref_val):
+    test = mock_covariance.mass_richness_integral(z, 0, remove_bias=True)
+    assert test == pytest.approx(ref_val)
 
-    test1 = cc_cov.mass_richness_integral(0.3, 0, remove_bias=True)
-    test2 = cc_cov.mass_richness_integral(0.35, 0, remove_bias=True)
 
-    np.testing.assert_almost_equal(ref1, test1)
-    np.testing.assert_almost_equal(ref2, test2)
-
-
-def test_double_bessel_integral():
+def test_double_bessel_integral(mock_covariance: ClusterCounts):
     ref = 8.427201745032292e-05
-    cc_cov = get_mock_covariance()
-    test = cc_cov.double_bessel_integral(0.3, 0.3)
-    np.testing.assert_almost_equal(ref, test)
+    test = mock_covariance.double_bessel_integral(0.3, 0.3)
+    assert test == pytest.approx(ref)
 
 
-def test_shot_noise():
+def test_shot_noise(mock_covariance: ClusterCounts):
     ref = 63973.635143644424
-    cc_cov = get_mock_covariance()
-    test = cc_cov.shot_noise(0, 0)
-    np.testing.assert_almost_equal(test, ref)
-
-
-def test_integral_mass():
-    cc_cov = get_mock_covariance()
-    ref1 = 2.596895139062984e-05
-    ref2 = 2.5910691906342223e-05
+    test = mock_covariance.shot_noise(0, 0)
+    assert test == pytest.approx(ref, 0.1)
```

Comment:
The relative difference was of order <1e-5 if I remember correctly. I do not think you need to put the threshold at 0.1 and i don't think this is worrisome.

Reply:
@mattkwiecien change this. The first argument of `pytest.approx` is the relative difference not the absolute difference. You should test that the relative difference is ~1e-5 or 1e-6

Reply:
Ah I understand now, I wasn't exactly clear what you meant before.